src/net/poll.rs

```diff
@@ -270,14 +267,15 @@ impl NetworkState {
             self.event_map.len(),
             self.event_capacity
         );
+
+        if let Err(e) = sock.shutdown(Shutdown::Both) {
+            warn!("Failed to shut down socket {}: {:?}", event_id, &e);
+        }
+
```

Comment:
Should failures in `sock.shutdown` or `poll.deregister` return an `Err` from this function?